exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -4,166 +4,193 @@
+            continueMoveFromStartOfBoard();
+        }
+
+        printPlayerPosition(playerID);
+    }
+
+    private void continueMoveFromStartOfBoard() {
+        playerPositions[currentPlayerID] = getPlayerPosition(currentPlayerID) - 12;
+    }
+
+    private boolean hasPlayerPassedFinalTile(int playerID) {
+        return getPlayerPosition(playerID) > 11;
+    }
+
+    private void askQuestion() {
+        if (getCategoryForPosition().equals(POP))
+            System.out.println(popQuestions.removeFirst());
+        if (getCategoryForPosition().equals(SCIENCE))
+            System.out.println(scienceQuestions.removeFirst());
+        if (getCategoryForPosition().equals(SPORTS))
+            System.out.println(sportsQuestions.removeFirst());
+        if (getCategoryForPosition().equals(ROCK))
+            System.out.println(rockQuestions.removeFirst());
+    }
+
+    private String getCategoryForPosition() {
+        return board[getPlayerPosition(currentPlayerID)];
+    }
+
+    private int getPlayerPosition(int playerID) {
+        return playerPositions[playerID];
+    }
+
+    public boolean handleCorrectAnswer() {
+        if (isPlayerInPenaltyBox(currentPlayerID)) {
+
+            if (isGettingOutOfPenaltyBox) {
+                boolean doesNotHaveSixCoins = addCoin(currentPlayerID);
+                switchToNextPlayer();
+
+                return doesNotHaveSixCoins;
+            } else {
+                switchToNextPlayer();
+                return true;
+            }
+
+        } else {
+            boolean doesNotHaveSixCoins = addCoin(currentPlayerID);
+            switchToNextPlayer();
+
+            return doesNotHaveSixCoins;
+        }
+    }
+
+    private void switchToNextPlayer() {
+        currentPlayerID++;
+        if (isLastPlayerInList()) resetBackToPlayerOne();
+    }
+
+    private boolean isPlayerInPenaltyBox(int playerID) {
+        return playersInPenaltyBox[playerID];
+    }
+
+    private void resetBackToPlayerOne() {
+        currentPlayerID = PLAYER_ONE_ID;
+    }
+
+    private boolean addCoin(int playerID) {
+        System.out.println("Answer was correct!!!!");
+        playerPurses[currentPlayerID]++;
+        System.out.println(players.get(playerID)
+                + " now has "
+                + getPlayerPurse(playerID)
+                + " Gold Coins.");
+
+        return playerDoesNotHaveSixCoins(currentPlayerID);
+    }
+
+    private int getPlayerPurse(int playerID) {
+        return playerPurses[playerID];
+    }
+
+    public boolean handleIncorrectAnswer() {
+        System.out.println("Question was incorrectly answered");
+        System.out.println(players.get(currentPlayerID) + " was sent to the penalty box");
+        playersInPenaltyBox[currentPlayerID] = true;
+
+        switchToNextPlayer();
+        return true;
+    }
+
+    private boolean isLastPlayerInList() {
+        return currentPlayerID == players.size();
+    }
+
+    private boolean playerDoesNotHaveSixCoins(int playerID) {

```

Comment:
Perhaps a bit nitpicky, but booleans tend to read a little better if they starts with "is" or "has". This method could read "doesPlayerHaveSixCoins"